packages/renderer/src/lib/preferences/LineWithLinks.svelte

```diff
@@ -0,0 +1,52 @@
+<script lang="ts">
+export let lineText: string;
+
+interface Segment {
+  text: string;
+  href?: string;
+}
+
+function createLink(text: string): Segment {
+  const match = text.match(/\[(?<text>.*)\]\[(?<url>.*)\]/);
+  return {
+    text: match.groups.text,
+    href: match.groups.url,
+  };
+}
+
+let text = lineText;
+let segments: Segment[] = [];
+while (text.length > 0) {
+  const match = text.search(/\[.*\]\[.*\]/);
+  if (match > 0) {
+    // it is a link
+    segments.push({ text: text.substring(0, match) });
+    const linkMatch = text.match(/\[(?<text>.*)\]\[(?<url>.*)\]/);
+    segments.push({
+      text: linkMatch.groups.text,
+      href: linkMatch.groups.url,
+    });
+    const linkOffcet = 4 + linkMatch.groups.text.length + linkMatch.groups.url.length;
+    text = text.substring(match + linkOffcet, text.length);
+  } else {
+    segments.push({ text });
+    text = '';
+  }
+  console.log(segments);
+}
+</script>
+
+<p>
+  {#if segments.length === 0}
```

Comment:
I think you can just use the #each without this check if you don't add anything. if it's empty it doesn't loop.
Then you can add a class to add a margin/padding/whatever with a clause in it
`class:m-5={segments.length === 0}`